File: AOJ/ALDS1_3_A/main.cpp
#include <bits/stdc++.h>

#include <stack>
using namespace std;

typedef long long ll;

int main() {
    string c;
    stack<int> s;
    while (cin >> c) {
        if (c == "+" or c == "-" or c == "*") {
            ll a = s.top();
            s.pop();
            ll b = s.top();
            s.pop();

            ll res;
            if (c == "+") res = b + a;
            if (c == "-") res = b - a;
            if (c == "*") res = b * a;
            s.push(res);
        } else {
            s.push(stoi(c));
        }
    }
    cout << s.top() << endl;
}

File: AOJ/ALDS1_3_A/main.py

L = input().split()

stack = []

for s in L:
    if s == '+' :
        a = stack.pop()
        b = stack.pop()
        stack.append(a+b)
    elif s == '-':
        a = stack.pop()
        b = stack.pop()
        stack.append(b-a)        
    elif s == '*':
        a = stack.pop()
        b = stack.pop()
        stack.append(a*b)
    else:
        stack.append(int(s))

print(stack[0])